src/svg/qsvggenerator.cpp

```diff
@@ -998,7 +1017,36 @@ void QSvgPaintEngine::updateState(const QPaintEngineState &state)
     if (d->afterFirstUpdate)
         *d->stream << "</g>\n\n";
 
-    *d->stream << "<g ";
+	QPainter* p = painter();
+	if (p->hasClipping()) {
+		std::string clip_path = "";
+		QPainterPath path = p->clipPath();
+
+		if (path.elementCount() > 0) {
+			QPainterPath::Element starting_point = path.elementAt(0);
+			clip_path.append("<path d=\"M " + std::to_string(starting_point.x) + " " + std::to_string(starting_point.y) + " ");
+
+			for (int i = 1; i < path.elementCount(); i++) {
+				QPainterPath::Element element = path.elementAt(i);
+				std::string point = "L" + std::to_string(element.x) + ", " + std::to_string(element.y) + " ";
+				clip_path.append(point);
+			}
+
+			bool key_exists = clip_path_to_id.count(clip_path);
+			if (!key_exists) {
+				clip_path_to_id[clip_path] = clip_counter++;
```

Comment:
FYI These two lines have different meaning:
`clip_path_to_id[clip_path] = clip_counter++;`
`clip_path_to_id[clip_path] = ++clip_counter;`
While I don't understand the purpose of `clip_counter` I cannot say whether what you doing is making sense. Please review the logic, whether `cli_path_to_id` supposed to contain previous value or updated one